Shopify/ruby-lsp
# Add basic inlay hint support

### Motivation

Part of https://github.com/Shopify/ruby-lsp/issues/55

Inlay hints are neat and it seemed like an easy issue to get started with. Started with adding a hint for the implied StandardError when a user makes a bare rescue.

### Implementation

- Largely similar to the other request types
- ~~Inlay Hints were added in LSP spec 3.17 so it's not as well-typed as I'd like. The gem is still at 3.16~~
- Checks for `node.is_a?(SyntaxTree::Rescue) && node.exception.nil?`

### Automated Tests

I added an expectation test, can certainly add more if necessary.

### Manual Tests

- I modified the gemfile of a local project (Tapioca) to point to my local copy of Ruby-LSP
- I modified the client to support inlay hints (see https://github.com/Shopify/vscode-ruby-lsp/pull/202)
- I opened Tapioca when running the client in debug mode so it was using the new LSP version and the client that could support it.
- I typed the following code to ensure the inlay hint popped up:

```ruby
begin
  puts "hello"
rescue
  puts "bye-bye"
end
```
